packages/react-components/src/organisms/Form.tsx

```diff
@@ -96,7 +96,9 @@ const Form = <T extends void | Record<string, unknown>>({
           onCancel,
           isSaving: status === 'isSaving',
           onSave: getWrappedOnSave(onSave),
-          onSaveDraft: getWrappedOnSave(onSaveDraft),
+          ...(onSaveDraft
```

Comment:
something we did for the modal, was instead of passing all the function we want wrapped, we just return the wrapper and we can wrap any function we want

Reply:
nice, I'll open a new PR with that soon